helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -219,4 +220,22 @@ public static String serializeByComma(List<String> objects) {
 
     return idealStateMap;
   }
+
+  /**
+   * Remove the given message from ZK using the given accessor. This function will
+   * not throw exception
+   * @param accessor HelixDataAccessor
+   * @param msg message to remove
+   * @param instanceName name of the instance on which the message sits
+   * @return true if success else false
+   */
+  public static boolean removeMessageFromZK(HelixDataAccessor accessor, Message msg,
+      String instanceName) {
+    try {
+      return accessor.removeProperty(msg.getKey(accessor.keyBuilder(), instanceName));
+    } catch (Exception e) {
```

Comment:
Do we throw NoNodeException here, if there is, NoNodeException should be treated as success?

Reply:
it will not. the reason I did a general try-catch here is because I want to keep removeProperty semantics (only return true/false) here, but we do have leaked exception in underlying implementations of removeProperty()